### PR TITLE
fix(ONYX-719): infinite loading on the notifications list

### PR DESCRIPTION
**Pull request by dariakoko**
This PR resolves [ONYX-719] 

### Description

This PR fixes jumpy `FlatList` and infinite loading on the ActivityList component
Weirdly enough removing the `flex={1}` property [here](https://github.com/artsy/eigen/commit/1efd2895478c7cd8dd0851def9f312319773cec2) was the solution 🙃 
In addition to the fix some refactoring has been made

Thanks @gkartalis for helping with this!

https://github.com/artsy/eigen/assets/36167539/28902d5d-bc49-455f-904d-0aadd388ef42

### PR Checklist

- [x] I have tested my changes on **iOS** and **Android**.
- [ ] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [ ] I have added **tests**, or my changes don't require any.
- [ ] I added an **[app state migration]**, or my changes do not require one.
- [ ] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [ ] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- fix jumpy FlatList and infinite loading on the ActivityList component - george & daria

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[ONYX-719]: https://artsyproduct.atlassian.net/browse/ONYX-719?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Cross-platform user-facing changes (fix jumpy FlatList and infinite loading on the ActivityList component - george & daria)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 327eb33011278fa1fe92a90c672fe0ec576f9883
</p>

File: src/app/Scenes/Activity/ActivityList.tsx [hunk 1]
```diff
@@ -84,28 +66,6 @@ export const ActivityList: React.FC<ActivityListProps> = ({ viewer, type, me })
...
-      <ScrollViewComponent
-        refreshControl={<RefreshControl onRefresh={handleRefresh} refreshing={refreshing} />}
-      >
-        <Flex flex={1} justifyContent="center" top={headerOffset}>
-          <ActivityEmptyView type={type} />
-        </Flex>
-      </ScrollViewComponent>
-    )
-  }
-
```
Comment (gkartalis) [inline src/app/Scenes/Activity/ActivityList.tsx:108]: 👋

File: src/app/Scenes/Activity/ActivityList.tsx [hunk 2]
```diff
@@ -134,7 +94,12 @@ export const ActivityList: React.FC<ActivityListProps> = ({ viewer, type, me })
         )
       }
       onEndReached={handleLoadMore}
-      renderItem={({ item }) => <>{item.content}</>}
+      renderItem={({ item }) => <ActivityItem notification={item} />}
+      ListEmptyComponent={
+        <Flex flex={1} justifyContent="center">
+          <ActivityEmptyView type={type} />
+        </Flex>
+      }
```
Comment (gkartalis) [inline src/app/Scenes/Activity/ActivityList.tsx:102]: 🎉

File: src/app/Scenes/Activity/ActivityItem.tsx [hunk 3]
```diff
@@ -92,7 +92,7 @@ export const ActivityItem: React.FC<ActivityItemProps> = (props) => {
                   )}
                 </Flex>
 
-                <Flex flex={1} justifyContent="center">
+                <Flex justifyContent="center">
```
Comment (gkartalis) [inline src/app/Scenes/Activity/ActivityItem.tsx:95]: THAT WAS IT BUT

**Review by gkartalis — APPROVED** [review]
awesome! 🤩

**Pull request merged**